kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KakfaServerExtensionTest.java

```diff
@@ -0,0 +1,149 @@
+        extension.destroy(server, registry);
+    }
+
+    @Test
+    public void testKafkaServerExecutorSignal() throws InterruptedException {
+        when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(
+                new Signal("MySignal", "MySignal", "String"))));
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+        extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+        publishEvent("MySignal", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"javierito\"}");
+        verify(processService, getTimeout()).signalEvent("MyDeploy", "MySignal", "javierito");
+    }
+
+    @Test
+    public void testKafkaServerExecutorMessage() throws InterruptedException {
+
+        Message msg = new Message("MyMessage");
+        msg.setName("Hello");
+        msg.setType("String");
+        when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(msg)));
+        extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+        publishEvent("Hello", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"pepe\"}");
+        verify(processService, getTimeout()).signalEvent("MyDeploy", "Message-Hello", "pepe");
+    }
+
+    @Test
+    public void testKafkaServerExecutorMessageTopic() throws InterruptedException {
+
+        final String topicProperty = KafkaServerExtension.TOPIC_PREFIX + "Hello";
+        System.setProperty(topicProperty, "MyTopic");
+        try {
+            Message msg = new Message("MyMessage");
+            msg.setName("Hello");
+            msg.setType("String");
+            when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(
+                    msg)));
+            extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+            publishEvent("MyTopic", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"pepe\"}");
+            verify(processService, getTimeout()).signalEvent("MyDeploy", "Message-Hello", "pepe");
+        } finally {
+            System.clearProperty(topicProperty);
+        }
+    }
+
+    private VerificationMode getTimeout() {
+        return timeout(Long.getLong(KafkaServerExtension.KAFKA_EXTENSION_PREFIX + "poll.interval", 1L) * 2000).times(1);
+    }
+
+    private void publishEvent(String topic, String cloudEventText) {
+        Set<String> topics = extension.getMockConsumer().subscription();
+        assertTrue("Topic " + topic + " not found", topics.contains(topic));
+        List<TopicPartition> partitions = Collections.singletonList(new TopicPartition(topic, 0));
+        extension.getMockConsumer().rebalance(partitions);
+        Map<TopicPartition, Long> partitionsBeginningMap = new HashMap<TopicPartition, Long>();
+        Map<TopicPartition, Long> partitionsEndMap = new HashMap<TopicPartition, Long>();
```

Comment:
Can be replaced with empty diamonds
```suggestion
        Map<TopicPartition, Long> partitionsBeginningMap = new HashMap<>();
        Map<TopicPartition, Long> partitionsEndMap = new HashMap<>();
```